iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/BaseHiveIcebergMetaHook.java

```diff
@@ -202,6 +203,65 @@ public void preCreateTable(CreateTableRequest request) {
     // Remove hive primary key columns from table request, as iceberg doesn't support hive primary key.
     request.setPrimaryKeys(null);
     setSortOrder(hmsTable, schema, catalogProperties);
+    setWriteMetadataCleanupProperties(hmsTable.getParameters());
+  }
+
+  private void setWriteMetadataCleanupProperties(Map<String, String> hmsParams) {
+
+    boolean isEnabled = isConfigEnabled(
+        TableProperties.METADATA_DELETE_AFTER_COMMIT_ENABLED,
+        HIVE_ICEBERG_METADATA_DELETE_AFTER_COMMIT_ENABLED_DEFAULT,
+        hmsParams);
+
+    if (isEnabled) {
+      catalogProperties.put(TableProperties.METADATA_DELETE_AFTER_COMMIT_ENABLED, true);
+
+      addCatalogProperty(
+          TableProperties.METADATA_PREVIOUS_VERSIONS_MAX,
+          TableProperties.METADATA_PREVIOUS_VERSIONS_MAX_DEFAULT,
+          hmsParams);
+    }
+  }
+
+  /*
+   * Determines whether a given configuration is enabled, following this precedence:
+   * 1. If the configuration exists in the HMS table parameters (hmsParams), use its boolean value.
+   * 2. Otherwise, check the session configuration (conf) and parse its boolean value if present.
+   * 3. If neither HMS parameters nor session configuration provide a value, return the specified defaultValue.
+   * This ensures that table-level settings override session settings, with a fallback to the default.
+   */
+  private boolean isConfigEnabled(String configName, boolean defaultValue, Map<String, String> hmsParams) {
```

Comment:
should this be moved to IcebergTableUtil. see `isFanoutEnabled`

Reply:
Done